app/components/primer/alpha/overlay.rb

```diff
@@ -0,0 +1,194 @@
+        Primer::Alpha::Overlay::Body.new(
+          padding: padding,
+          **system_arguments
+        )
+      }
+
+      # Footer content.
+      #
+      # @param show_divider [Boolean] Show a divider between the footer and body.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :footer, "Footer"
+
+      # @example Overlay with Cancel and Submit buttons
+      #   @description
+      #     An ID is provided which enables wiring of the open and close buttons to the Overlay.
+      #   @code
+      #     <%= render(Primer::Alpha::Overlay.new(
+      #       title: "Overlay Example",
+      #       id: "my-Overlay",
+      #       role: :dialog,
+      #     )) do |d| %>
+      #       <% d.with_show_button { "Show Overlay" } %>
+      #       <% d.with_body do %>
+      #         <p>Some content</p>
+      #       <% end %>
+      #     <% end %>
+      # @param id [String] The id of the Overlay.
+      # @param title [String] Describes the content of the Overlay.
+      # @param subtitle [String] Provides dditional context for the Overlay, also setting the `aria-describedby` attribute.
+      # @param popover [Symbol] The popover behaviour. <%= one_of(Primer::Alpha::Overlay::POPOVER_OPTIONS) %>
+      # @param popover [Symbol] The popover behaviour. <%= one_of(Primer::Alpha::Overlay::POPOVER_OPTIONS) %>
+      # @param size [Symbol] The size of the Overlay. <%= one_of(Primer::Alpha::Overlay::SIZE_OPTIONS) %>
+      # @param padding [Symbol] The padding given to the Overlay body. <%= one_of(Primer::Alpha::Overlay::PADDING_OPTIONS) %>
+      # @param anchor [String] An ID of the element to anchor onto. Defaults to the `show_button`.
+      # @param anchor_align [Symbol] The anchor alignment of the Overlay. <%= one_of(Primer::Alpha::Overlay::ANCHOR_ALIGN_OPTIONS) %>
+      # @param anchor_side [Symbol] The side to anchor the Overlay to. <%= one_of(Primer::Alpha::Overlay::ANCHOR_SIDE_OPTIONS) %>
+      # @param anchor_offset [Symbol] The anchor offset to give the Overlay. <%= one_of(Primer::Alpha::Overlay::ANCHOR_OFFSET_OPTIONS) %>
+      # @param allow_out_of_bounds [Boolean] Allow the Overlay to overflow its container.
+      # @param visually_hide_title [Boolean] If true will hide the heading title, while still making it available to Screen Readers.
+      # @param role [String] The ARIA role. <%= one_of(Primer::Alpha::Overlay::ROLE_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      def initialize(
+        title:,
+        role:,
+        subtitle: nil,
+        popover: DEFAULT_POPOVER,
+        size: DEFAULT_SIZE,
+        padding: DEFAULT_PADDING,
+        anchor: nil,
+        anchor_align: DEFAULT_ANCHOR_ALIGN,
+        anchor_offset: DEFAULT_ANCHOR_OFFSET,
+        anchor_side: DEFAULT_ANCHOR_SIDE,
+        allow_out_of_bounds: false,
+        visually_hide_title: false,
+        id: "overlay-#{(36**3 + rand(36**4)).to_s(36)}",
```

Comment:
We have a function for this https://github.com/primer/view_components/blob/462bfb63e83d6a3535cd68c3c9b2176bb5e27717/app/components/primer/component.rb#L33-L35